### ISSUE TITLE
Merge the two generate md table functions

### ISSUE THREAD

**Issue by mutterroland** [Akhliskun/firefox-infra-changelog#101]
Currently we have two functions that generate markdown tables. After refactoring filter_hg_commit_data it looks like the two functions that generate md tables can be merged into one.

I'll start by looking into those 2 functions 1:1 , compare them , modify if necessary , test the code and make a PR . I'm going to self assign this one.
## Requirements
* Don't use classes
* Runs on Py3.X
* Document the code

LATER EDIT: Saw that there are two more functions that are basically the same and can be merged. Those two are the extract ones . Also, generate_main_md_table can can be simplified as well.

**Comment by mutterroland** [comment]
While merging the two create_md_table functions,  I saw that extract_json functions can also be merged into one. Basically, I added path_to_files parameter for each set of functions and used that to save ~90 lines of code. Previously we had sets of functions for each main action required either for Git or Hg . After modifying the way we generate HG part in #96 I concluded that , basically, we store information the same way for Hg as Git is. So ... why not use that in a proper way ?
